### PR TITLE
refactor: plugins enable requirements

### PR DESCRIPTION
**Pull request by guanbinrui**
## Description

Closes MF-0000

## Type of change

- [ ] Documentation
- [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

[MF-5266]: https://mask.atlassian.net/browse/MF-5266?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/plugin-infra/src/manager/dashboard.ts [hunk 1]
```diff
@@ -27,5 +28,16 @@ export function useActivatedPluginDashboard(pluginID: string) {
...
+    startDaemon(host, (id) => {
+        const def = getPluginDefine(id)
+        if (!def) return false
+
+        const currentNetwork = getSiteType()
+        if (!currentNetwork) return false
+
+        const status = def.enableRequirement.supports.sites[currentNetwork]
+        if (def.enableRequirement.supports.type === 'opt-in' && status !== true) return false
+        if (def.enableRequirement.supports.type === 'opt-out' && status === true) return false
```
Comment (Jack-Works) [inline packages/plugin-infra/src/manager/dashboard.ts]: you should not do this check in the dashboard. dashboard plugin host only start in one place, which is dashboard, and it does not need this type of check.

File: packages/mask/src/extension/popups/normal-client.tsx [hunk 2]
```diff
@@ -19,7 +18,6 @@ if (
 ) {
     console.time('[SSR] Fill data')
     await Promise.all([
-        activateSiteAdaptorUI(),
```
Comment (Jack-Works) [inline packages/mask/src/extension/popups/normal-client.tsx]: why did you remove this? if you remove this, where do you start the site adaptor?

Comment (guanbinrui) [reply packages/mask/src/extension/popups/normal-client.tsx]: Why do we need to start site adaptor on popups page?

Comment (Jack-Works) [reply packages/mask/src/extension/popups/normal-client.tsx]: oh, we're expected to not start this (we used to start it because some plugin crashes), if there is no bug with this, I'm okay with this.
